### PR TITLE
Added login logic, logout logic, and integrated backend data

### PR DESCRIPTION
**Pull request by AsaturHakhverdyan**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,25 +1,41 @@
...
+
+useEffect(() => {
+  (async () => {
+    if(localStorage.getItem("jwt_token")){
+      navigate("/home")
+    }else{
+      navigate("/login")
+    }
+  })()
+},[]);
```
Comment (hrachocode) [inline src/App.tsx]:
```suggestion
useEffect(() => {
  (async () => {
    if(!localStorage.getItem("jwt_token")){
      navigate("/login")
    }
  })()
},[]);
```

**Review by hrachocode — CHANGES_REQUESTED** [review]

File: src/Components/home/HomePage.tsx [hunk 2]
```diff
@@ -1,56 +1,100 @@
...
 import ProductItem from "./ProductItem";
-import NavbarInfo from "../header/NavbarInfo";
 import InformationBlock from "./components/InformationBlock";
 import Pagination from "./components/Pagination";
+import { useSearchParams } from "react-router-dom";
 
 const HomePage: React.FC = () => {
-  const [productsData, setProductsDataData] = useState<Products[]>(productItems);
-  const [selectedData, setSelectedData] = useState<Products[]>([]);
+  const BASE_URL: string = process.env.REACT_APP_BASE_URL as string;
```
Comment (IshkhanDavtyan) [inline src/Components/home/HomePage.tsx]: use one of these types ( :string or as string)

File: src/Components/home/components/InformationBlock.tsx [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
-import { productItems } from "../base";
 
-const InformationBlock = ({selectedDatalength}:{selectedDatalength:number}) => {
+const InformationBlock = ({
+  totolCount,
+  productPerPage,
+}: {
+  totolCount: number;
+  productPerPage:number,
+}) => {
```
Comment (IshkhanDavtyan) [inline src/Components/home/components/InformationBlock.tsx:8]: Create interface for this types

File: src/Components/home/components/Pagination.tsx [hunk 4]
```diff
@@ -1,68 +1,68 @@
...
+  currentPage,
+  productPerPage,
+  setCurrentPage,
 }: {
-  productsData: Products[],
-  setSelectedData:  React.Dispatch<React.SetStateAction<Products[]>>,
+  totolCount: number;
+  currentPage: number;
+  productPerPage: number;
+  setCurrentPage: React.Dispatch<React.SetStateAction<number>>;
```
Comment (IshkhanDavtyan) [inline src/Components/home/components/Pagination.tsx:13]: Create interface for this types

File: src/Components/home/components/Pagination.tsx [hunk 5]
```diff
@@ -1,68 +1,68 @@
...
-    setSelectedData(productsData.slice(indexOfFirstProcuct,indexOfLastProduct));
-  },[currentPage]);
-  
-  const paginationCount = [];
-  for (let i = 1; i < Math.ceil(productsData.length / productPerPage); i++) {
-    paginationCount.push(i);
+  const paginationCount: number[] = [];
+  for (let i = 0; i < Math.ceil(totolCount / productPerPage); i++) {
+    paginationCount.push(i + 1);
+  }
```
Comment (IshkhanDavtyan) [inline src/Components/home/components/Pagination.tsx]: get Array of pages like this
 const paginationCount=Array.from({length: Math.ceil(totalCount/productPerPage)}, (_, i) => i + 1)

File: src/Components/home/components/Pagination.tsx [hunk 6]
```diff
@@ -1,68 +1,68 @@
...
-  const onChangePagination = (number: number) => {
-   setCurrentPage(number);
-};
+  const decrementPaginationCount = () => {
+    if (currentPage === paginationCount.length) {
+      return false;
+    } else {
+      setCurrentPage(currentPage + 1);
+    }
+  };
```
Comment (IshkhanDavtyan) [inline src/Components/home/components/Pagination.tsx:32]: create utils.ts file inside component direction and move these functions there (decrementPaginationCount,incrementPaginationCount)

File: src/Components/home/components/Product.tsx [hunk 7]
```diff
@@ -1,15 +1,140 @@
...
+      } finally {
+        setIsloading(false);
+      }
+    })();
+  }, [BASE_URL, id]);
+
+  const addToFavorites = () => {
+    console.log("there will be favorite products call");
+  };
+
```
Comment (IshkhanDavtyan) [inline src/Components/home/components/Product.tsx]: cool :) , this function will be moved in utils.ts file

File: src/Components/leftbar/LeftBarMenu.tsx [hunk 8]
```diff
@@ -1,34 +1,72 @@
...
+import axios from "axios";
+import { AttributeCategory, Attributes } from "../../utils/interface";
 import FillterTitle from "./components/FillterTitle";
 import LeftBarInfo from "./components/LeftBarInfo";
 import LeftBarItem from "./components/LeftBarItem";
+import { FiltersItemList } from "../../utils/interface";
 
-const LeftBarMenu  = () => {
+const LeftBarMenu = () => {
+  const BASE_URL: string = process.env.REACT_APP_BASE_URL as string;
```
Comment (IshkhanDavtyan) [inline src/Components/leftbar/LeftBarMenu.tsx]: use (:string or as string) one of types , these two are equivalent

File: src/Components/leftbar/LeftBarMenu.tsx [hunk 9]
```diff
@@ -1,34 +1,72 @@
...
+      setFilterItemList(filters);
     } else {
-      return setFilterItemList([...filterItemList, filtersItem]);
+      filters.push({
+        name: filtersItemNmae,
+        description: filtersItemDes,
+      });
+      setFilterItemList(filters)
     }
   };
```
Comment (IshkhanDavtyan) [inline src/Components/leftbar/LeftBarMenu.tsx:53]: move this function to utils.ts file

File: src/Components/leftbar/components/LeftBarInfo.tsx [hunk 10]
```diff
@@ -1,53 +1,62 @@
...
 
-const LeftBarInfo = ({filterItemList , setFilterItemList}:{filterItemList:string[], setFilterItemList:React.Dispatch<React.SetStateAction<string[]>>}) => {
-  
+const LeftBarInfo = ({
+  filterItemList,
+  setFilterItemList,
+}: {
+  filterItemList: FiltersItemList[];
+  setFilterItemList: React.Dispatch<React.SetStateAction<FiltersItemList[]>>;
+}) => {
```
Comment (IshkhanDavtyan) [inline src/Components/leftbar/components/LeftBarInfo.tsx:16]: Create interface for this types

File: src/Components/leftbar/components/LeftBarInfo.tsx [hunk 11]
```diff
@@ -1,53 +1,62 @@
...
   const ClearHandler = (elem: string) => {
-     const newDataList = filterItemList.filter(item => item !== elem)
-     setFilterItemList(newDataList);
+    const newDataList = filterItemList.filter((item) => item.name !== elem);
+    setFilterItemList(newDataList);
   };
```
Comment (IshkhanDavtyan) [inline src/Components/leftbar/components/LeftBarInfo.tsx:68]: move this function to utils.ts file

File: src/Components/leftbar/components/LeftBarInfo.tsx [hunk 12]
```diff
@@ -1,53 +1,62 @@
...
-              onClick={() => ClearHandler(elem)}
-            />
+            <div className="flex items-center">
+              {elem.description.includes("https://") && (
+                <div>
+                  <img
+                    src={elem.description}
+                    alt="#"
+                    className="w-[20px] "
+                  />
```
Comment (IshkhanDavtyan) [inline src/Components/leftbar/components/LeftBarInfo.tsx:88]: alt will be empty string if there is nothing to  describe image

File: src/Components/leftbar/components/LeftBarItem.tsx [hunk 13]
```diff
@@ -1,31 +1,63 @@
...
+  categories,
+  filterItems_name,
+  filterItemList,
+  onChangeHandler,
+}: {
+  categories: AttributeCategory[];
+  filterItems_name: string;
+  filterItemList: FiltersItemList[];
+  onChangeHandler: (filtersItemName: string, filtersItemDes: string) => void;
+}) => {
```
Comment (IshkhanDavtyan) [inline src/Components/leftbar/components/LeftBarItem.tsx:15]: Create interface

File: src/Components/login/LoginForm.tsx [hunk 14]
```diff
@@ -5,98 +5,94 @@ import axios from "axios";
 import Photo from "../../images/Rectangle.svg";
 
 const LoginForm: React.FC = () => {
+  const LOGIN_URL: string = process.env.REACT_APP_LOGIN_URL as string;
```
Comment (IshkhanDavtyan) [inline src/Components/login/LoginForm.tsx]: use one of type declaration (:string or as string)

File: src/Components/login/LoginForm.tsx [hunk 15]
```diff
@@ -5,98 +5,94 @@ import axios from "axios";
...
   const navigate = useNavigate();
-  console.log(navigate);
-  
   const [username, setUsername] = useState<string>("");
   const [userpassword, setUserpassword] = useState<string>("");
   const [showpassword, setShowpassword] = useState<boolean>(false);
 
-  
-  
   const formHendler = async (event: React.SyntheticEvent<HTMLFormElement>) => {
```
Comment (IshkhanDavtyan) [inline src/Components/login/LoginForm.tsx:14]: move this function to utils.ts file

File: src/Components/update/UpdatePassword.tsx [hunk 16]
```diff
@@ -1,102 +1,138 @@
...
-      if(password1 === password2) {
-         setOpenErrorDiv(false);
-         setPassword1('');
-         setPassword2('');
-      }
-      else if(password1 !== password2) {
-         setOpenErrorDiv(true);
-      }
-   };
+  const clickHandler = (event: React.SyntheticEvent<HTMLFormElement>) => {
```
Comment (IshkhanDavtyan) [inline src/Components/update/UpdatePassword.tsx:22]: move to utils.ts file

**Review by IshkhanDavtyan — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 17]
```diff
@@ -1,25 +1,39 @@
...
 import LoginEmailPass from "./Components/update/LoginEmailPass";
 import UpdatePassword from "./Components/update/UpdatePassword";
+import Layout from "./layout/Layout";
 
 function App() {
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    (async () => {
+      if (!localStorage.getItem("jwt_token")) {
```
Comment (hrachocode) [inline src/App.tsx]:
```suggestion
      if (!localStorage.getItem(LOCAL_STORAGE_KEYS.JWT_TOKEN)) {
```

export const LOCAL_STORAGE_KEYS = {
  JWT_TOKEN: "jwt_token",
  JWT_REFRESH_TOKEN: "jwt_refresh_token"
}

File: src/App.tsx [hunk 18]
```diff
@@ -1,25 +1,39 @@
...
+        navigate("/login");
```
Comment (hrachocode) [inline src/App.tsx]:
```suggestion
        navigate(PAGES.LOGIN);
```

File: src/Components/header/Header.tsx [hunk 19]
```diff
@@ -1,37 +1,89 @@
...
-  const [ search, setSearch ] = useState<string>('')
-  
+  const navigate = useNavigate();
+  const [search, setSearch] = useState<string>("");
+  const [userInfo, setUserInfo] = useState<string>("");
+  const [searchProduct, setSearchProduct] = useState<SingleProducts[]>();
+
+  const searchHandleSubmit = (e: React.SyntheticEvent<HTMLFormElement>) => {
+    e.preventDefault();
+  };
```
Comment (hrachocode) [inline src/Components/header/Header.tsx]:
```suggestion

```

File: src/Components/header/Header.tsx [hunk 20]
```diff
@@ -1,37 +1,89 @@
...
-    <div className="py-9 px-4 bg-[#004f87]">
-      <div className="block md:flex  items-center justify-between">
-        <div>
-          <h1 className="text-2xl md:text-4xl tracking-[.2rem] md:tracking-[-.2rem] mb-1 text-white/40">
-           here will be user info
-          </h1>
+    <div className="py-[20px]  px-2 bg-gray-300 mb-[30px]">
+      <div className="block md:flex justify-between items-center relative">
+        <div className="flex items-center">
+          <div className="">
```
Comment (hrachocode) [inline src/Components/header/Header.tsx]:
```suggestion
          <div>
```

File: src/Components/header/Header.tsx [hunk 21]
```diff
@@ -1,37 +1,89 @@
...
+            <img src={logo} alt="" />
```
Comment (hrachocode) [inline src/Components/header/Header.tsx]:
```suggestion
            <img src={logo} alt="logo" />
```

File: src/Components/header/Header.tsx [hunk 22]
```diff
@@ -1,37 +1,89 @@
...
+              onChange={(e) => setSearch(e.target.value)}
+            />
+            <AiOutlineSearch
+              size={20}
+              className="absolute right-2 text-gray-500"
+            />
+          </form>
+          <div
+            className={
+              search.length !== 0
```
Comment (hrachocode) [inline src/Components/header/Header.tsx]:
```suggestion
              !!search.length
```

File: src/Components/header/components/ChooseComponent.tsx [hunk 23]
```diff
@@ -13,29 +13,29 @@ const ChooseComponent = ({props,title}: {props: string[],title: string}) => {
...
-          <p className="text-[#094570] font-[600] text-[10px]  uppercase md:text-[13px]">
+          <p className="text-[#094570] font-[600] text-[10px]  uppercase md:text-[12px]">
             {title}:
           </p>
         </div>
         <div>
           <h1
-            className="w-[150px] border-2 border-sky-200 rounded-xl px-3 ml-2 text-[10px] cursor-pointer relative"
+            className="w-[100px] xs:w-[150px] border-2 border-sky-200 rounded-xl px-3 ml-2 text-[10px] cursor-pointer relative"
             onClick={() => setShow(!show)}
```
Comment (hrachocode) [inline src/Components/header/components/ChooseComponent.tsx]:
```suggestion
            onClick={() => toggleShow(!show)}
```

File: src/Components/header/components/SearchProduct.tsx [hunk 24]
```diff
@@ -0,0 +1,117 @@
...
+}: {
+  search: string;
+  searchProduct: SingleProducts[] | undefined;
+  setSearchProduct: React.Dispatch<
+    React.SetStateAction<SingleProducts[] | undefined>
+  >;
+  setSearch: React.Dispatch<React.SetStateAction<string>>;
+}) => {
+  const [searchCatch, setSearchCatch] = useState<CatchError>();
+  const [isLoading, setisLoading] = useState<boolean>(false);
```
Comment (hrachocode) [inline src/Components/header/components/SearchProduct.tsx]:
```suggestion
  const [isLoading, setIsLoading] = useState<boolean>(false);
```

File: src/Components/header/components/SearchProduct.tsx [hunk 25]
```diff
@@ -0,0 +1,117 @@
...
+
+  useEffect(() => {
+    setisLoading(true);
+    const searchTime = setTimeout(async () => {
+      try {
+        const correncSearch = await axios.get(
+          `${process.env.REACT_APP_BASE_URL}/wc/v3/products?search=${search}&consumer_key=ck_91dcee2c1dda0dee993904bc2e5f59fe6be3895d&consumer_secret=cs_5c79352efe06c166099b85d537767a28afe619e5`
```
Comment (hrachocode) [inline src/Components/header/components/SearchProduct.tsx]: use process.env here

File: src/Components/header/components/SearchProduct.tsx [hunk 26]
```diff
@@ -0,0 +1,117 @@
...
+        );
+        if (correncSearch.data) {
+          setSearchProduct(correncSearch.data);
+        }
+      } catch (error: any | undefined) {
+        setSearchCatch(error);
+      } finally {
+        setisLoading(false);
+      }
+    }, 700);
```
Comment (hrachocode) [inline src/Components/header/components/SearchProduct.tsx]:
```suggestion
    }, UNMOUNT_TIMEOUT_VALUE);
```

File: src/Components/header/components/SearchProduct.tsx [hunk 27]
```diff
@@ -0,0 +1,117 @@
...
+            color={"cyan"}
+            loading={isLoading}
+            size={40}
+            aria-label="Loading Spinner"
+            data-testid="loader"
+          />
+        </div>
+      ) : (
+        <div>
+          {!searchProduct?.length && (
```
Comment (hrachocode) [inline src/Components/header/components/SearchProduct.tsx:35]:
```suggestion
          {!searchProduct?.length ? (
```

File: src/Components/header/components/SearchProduct.tsx [hunk 28]
```diff
@@ -0,0 +1,117 @@
...
+            <div className="w-[300px] md:w-[450px] flex items-center justify-center">
+              <h1 className="">Ոչինչ չի գտնվել</h1>
+            </div>
+          )}
+          {searchProduct?.map((product, index) => (
+            <Link to={`/product/${product.id}`} key={index}>
```
Comment (hrachocode) [inline src/Components/header/components/SearchProduct.tsx]:
```suggestion
            <Link to={`${PAGES.PRODUCT}/${product.id}`} key={index}>
```

File: src/Components/header/components/SearchProduct.tsx [hunk 29]
```diff
@@ -0,0 +1,117 @@
...
+              <div
+                className="w-[300px] md:w-[450px] flex items-center border border-cyan-100 my-[5px] cursor-pointer hover:scale-105 duration-100"
+                onClick={() => setSearch("")}
+              >
+                <div className="rounded-full">
+                  <img
+                    src={product.images[0].src}
```
Comment (hrachocode) [inline src/Components/header/components/SearchProduct.tsx]:
```suggestion
                    src={product?.images?.[0]?.src}
```

File: src/Components/header/components/SearchProduct.tsx [hunk 30]
```diff
@@ -0,0 +1,117 @@
...
+                    alt=""
+                    className="w-[100px] h-[100px] md:w-[180px] md:h-auto object-contain"
+                  />
+                </div>
+                <div className="ml-[20px]">
+                  <div>
+                    <p className="text-[12px] text-gray-400">{product.sku}</p>
```
Comment (hrachocode) [inline src/Components/header/components/SearchProduct.tsx]:
```suggestion
                    <p className="text-[12px] text-gray-400">{product?.sku}</p>
```

File: src/Components/header/components/SearchProduct.tsx [hunk 31]
```diff
@@ -0,0 +1,117 @@
...
+                    <h1 className="text-[14px]">{product.name}</h1>
```
Comment (hrachocode) [inline src/Components/header/components/SearchProduct.tsx]:
```suggestion
                    <h1 className="text-[14px]">{product?.name}</h1>
```

File: src/Components/home/HomePage.tsx [hunk 32]
```diff
@@ -1,55 +1,132 @@
...
 
-  //     if(correctData){
-  //       setData(correctData.data)
-  //     }
-  //   })()
-  // },[setData]);
+  const [productsData, setProductsData] = useState<Products[]>([]);
+  const [URL_PARAMS, setURL_PARAMS] = useState<string>("");
+  const [totolCount, setTotalCount] = useState<number>(0);
+  const [productPerPage] = useState<number>(9);
```
Comment (hrachocode) [inline src/Components/home/HomePage.tsx]: this is constant

File: src/Components/home/HomePage.tsx [hunk 33]
```diff
@@ -1,55 +1,132 @@
...
+  const PRODUCT = process.env.REACT_APP_PRODUCT as string;
 
-  //     if(correctData){
-  //       setData(correctData.data)
-  //     }
-  //   })()
-  // },[setData]);
+  const [productsData, setProductsData] = useState<Products[]>([]);
+  const [URL_PARAMS, setURL_PARAMS] = useState<string>("");
+  const [totolCount, setTotalCount] = useState<number>(0);
```
Comment (hrachocode) [inline src/Components/home/HomePage.tsx]: this is contant?

File: src/Components/home/components/Pagination.tsx [hunk 34]
```diff
@@ -1,68 +1,66 @@
...
-                to="#"
-                aria-current="page"
-                className="border-b mx-[10px] px-[3px] border-[#091852] text-sm font-medium text-[#091852] cursor-pointer"
-              >
-                {number}
-              </Link>
-            ))}
-            <AiOutlineArrowRight size={13}/>
-          </nav>
+          {paginationCount.length > 1 && (
```
Comment (hrachocode) [inline src/Components/home/components/Pagination.tsx]:
```suggestion
          {!!paginationCount.length && (
```

File: src/Components/home/components/Product.tsx [hunk 35]
```diff
@@ -1,15 +1,195 @@
...
+import "swiper/css/navigation";
+import SameProducts from "./SameProducts";
 
 const Product = () => {
+  const BASE_URL = process.env.REACT_APP_BASE_URL as string;
+  const CONSUMER_KEY = process.env.REACT_APP_CONSUMER_KEY as string;
+  const WC_V3 = process.env.REACT_APP_WC_V3 as string;
+
+  const [singleProduct, setSingleProduct] = useState<SingleProducts>();
+  const [sameProducts, setSameProducts] = useState<SingleProducts[] | undefined>();
```
Comment (hrachocode) [inline src/Components/home/components/Product.tsx]:
```suggestion
  const [similiarProducts, setSimiliarProducts] = useState<SingleProducts[] | undefined>();
```

File: src/Components/home/components/Product.tsx [hunk 36]
```diff
@@ -1,15 +1,195 @@
...
+  const [isLoading, setIsloading] = useState<boolean>(false);
+  const [Loading, setloading] = useState<boolean>(false);
```
Comment (hrachocode) [inline src/Components/home/components/Product.tsx]: names

File: src/Components/home/components/Product.tsx [hunk 37]
```diff
@@ -1,15 +1,195 @@
...
+        if (correctData) {
+          setSingleProduct(correctData.data);
+        }
+      } catch (err: any | undefined) {
+        setIsError(err);
+      } finally {
+        setIsloading(false);
+      }
+    })();
+  }, [BASE_URL, id, CONSUMER_KEY, WC_V3]);
```
Comment (hrachocode) [inline src/Components/home/components/Product.tsx]: create custom hook

Comment (hrachocode) [reply src/Components/home/components/Product.tsx]: because I saw a repeating code.

File: src/Components/home/components/Product.tsx [hunk 38]
```diff
@@ -1,15 +1,195 @@
...
+    setloading(true);
+    const productsIds = singleProduct?.related_ids.toString();
+    (async () => {
+      const products = await axios.get(
+        `${BASE_URL}/${WC_V3}/products/?include=${productsIds}&${CONSUMER_KEY}`
+      );
+      if (products) {
+        setSameProducts(products.data);
+        setloading(false);
+      }
```
Comment (hrachocode) [inline src/Components/home/components/Product.tsx]: try catch finally

File: src/Components/home/components/Product.tsx [hunk 39]
```diff
@@ -1,15 +1,195 @@
...
+    })();
+  }, [BASE_URL, singleProduct?.related_ids, CONSUMER_KEY, WC_V3]);
+
+  const addToFavorites = () => {
+    console.log("there will be favorite products call");
+  };
```
Comment (hrachocode) [inline src/Components/home/components/Product.tsx]:
```suggestion

```

File: src/Components/home/components/Product.tsx [hunk 40]
```diff
@@ -1,15 +1,195 @@
...
+
+  useEffect(() => {
+    setDescription(singleProduct?.description);
+    setShort_description(singleProduct?.short_description);
+  }, [singleProduct?.description, singleProduct?.short_description]);
```
Comment (hrachocode) [inline src/Components/home/components/Product.tsx]:
```suggestion

```

File: src/Components/home/components/Product.tsx [hunk 41]
```diff
@@ -1,15 +1,195 @@
...
+            <Swiper
+              spaceBetween={10}
+              centeredSlides={true}
+              autoplay={false}
+              pagination={{
+                clickable: true,
+              }}
+              navigation={true}
+              modules={[Autoplay, Pagination, Navigation]}
+              className="mySwiper"
```
Comment (hrachocode) [inline src/Components/home/components/Product.tsx]:
```suggestion
              {...defaultSLiderSettings}
              loop={true}
```

File: src/Components/home/components/Product.tsx [hunk 42]
```diff
@@ -1,15 +1,195 @@
...
+                    {singleProduct?.regular_price} AMD
+                  </p>
+                )}
+              </div>
+            </div>
+            <div className="mt-10">
+              <div className="mt-5 lg:mt-10 lg:flex justify-between  items-center">
+                <div>
+                  <div
+                    onClick={() => console.log("")}
```
Comment (hrachocode) [inline src/Components/home/components/Product.tsx]:
```suggestion
                    ```

File: src/Components/home/components/SameProducts.tsx [hunk 43]
```diff
@@ -0,0 +1,105 @@
...
+      }
+      if (+window.innerWidth  < 1024 && +window.innerWidth  > 768){
+        count = 4;
+      }
+      if (+window.innerWidth  < 768 && +window.innerWidth  > 640){
+        count = 2;
+      }
+      if (+window.innerWidth  < 480){
+        count = 1;
+      }
```
Comment (hrachocode) [inline src/Components/home/components/SameProducts.tsx]: use constants with defined numbers

File: src/Components/leftbar/LeftBarMenu.tsx [hunk 44]
```diff
@@ -1,40 +1,107 @@
...
 
-const LeftBarMenu  = () => {
+const LeftBarMenu = ({
+  pricesRange,
+  setPricesRange,
+}: {
+  pricesRange: { [key: string]: number | null }[];
+  setPricesRange: Dispatch<SetStateAction<{ [key: string]: number | null }[]>>;
+}) => {
+  const BASE_URL = process.env.REACT_APP_BASE_URL as string;
```
Comment (hrachocode) [inline src/Components/leftbar/LeftBarMenu.tsx]: export this from one file

**Review by hrachocode — CHANGES_REQUESTED** [review]

File: src/Components/home/components/SingleProduct.tsx [hunk 45]
```diff
@@ -173,23 +106,23 @@ const Product = () => {
...
             <div className="flex justify-between">
-              <div dangerouslySetInnerHTML={{ __html: short_description }} />
+              <div dangerouslySetInnerHTML={{ __html: singleProduct.short_description }} />
             </div>
           ) : null}
         </div>
       )}
       <div>
-        <SameProducts sameProducts={sameProducts} Loading={Loading} />
+        <SimilarProducts similarProducts={similarProducts} SimilarLoading={SimilarLoading} />
```
Comment (hrachocode) [inline src/Components/home/components/SingleProduct.tsx]: similarLoading

File: src/Components/home/hooks/useBaseProductsCall.tsx [hunk 46]
```diff
@@ -0,0 +1,65 @@
...
+    >
+  >
+) {
+  const [productsData, setProductsData] = useState<Products[]>([]);
+  const [isLoading, setIsloading] = useState<boolean>(false);
+  const [searchParams] = useSearchParams();
+  const [isError, setIsError] = useState<CatchError>();
+  const productPerPage: number = 9;
+  const totolCountRef = useRef(0);
+  const MinPriceQuery = searchParams.get("min_price") || "5000";
```
Comment (hrachocode) [inline src/Components/home/hooks/useBaseProductsCall.tsx]:
```suggestion
  const MinPriceQuery = searchParams.get(QUERY_PARAMS.MIN_PRICE) || "5000";
```

**Review by hrachocode — CHANGES_REQUESTED** [review]

**Review by hrachocode — APPROVED** [review]

**Pull request merged**